### ISSUE TITLE
Deploy hangs with node_modules

### ISSUE THREAD

**Issue by Stephen-Meyerhofer** [apex/apex#570]
- __Operating system__: macOS Sierra 10.12
  - __Apex Version__: 0.10.3

Apex is having trouble compressing my functions with a node_modules folder. I am not sure if it has to do with the npm-shrinkwrap.json file, I have had mixed results. It may be that deploy only works when there are changes to be made.

The node_modules folder is 5.9MB. It contains dependencies from installing 1 dependency, `npm i requestretry`.

Anyone else experiencing this? Any workarounds? Will a new release be out soon that contains already completed fixes?

I also occasionally get this nice crash log.

[crash.txt](https://github.com/apex/apex/files/528198/crash.txt)

**Comment by DavidTPate** [comment]
@Stephen-Meyerhofer I suspect this is a duplicate of #551 if you want to take a look at that.

**Comment by Stephen-Meyerhofer** [comment]
Yes, this is a duplicate of #551. However, I searched for "node_modules" and #551 did not appear. Should have searched for Sierra.

A couple of questions:

This indicates failure.
> go/work/src/github.com/apex/apex $ make
> [+] releasing 
> [+] re-generating
> boot/boilerplate/boilerplate.go:1: running "go-bindata": exec: "go-bindata": executable file not found in $PATH
> docs/docs.go:1: running "go-bindata": exec: "go-bindata": executable file not found in $PATH
> mock/mock.go:1: running "mockgen": exec: "mockgen": executable file not found in $PATH
> shim/shim.go:1: running "go-bindata": exec: "go-bindata": executable file not found in $PATH
> make: *** [release] Error 1

For those of us who are beginners at golang, what should we do to make `make` work?

Even though `make` failed, I seem to have successfully built my own binary by running ` go get -u ./...` and `go install ./...` (Not sure if running both was required)

What does `./...` mean?

I ran `cp go/work/bin/apex /usr/local/bin/apex` and `apex deploy` now works!

**Comment by polds** [comment]
`./...` is a go shortcut to mean "recursively go through the packages and sub-packages of the project starting in this directory.

One thing about go get is it basically does this in order:
```
git clone
go install
```

So a go get and go install is generally not needed (but no harm doing it)
